src/browser-driver-manager.js

```diff
@@ -1,259 +1,259 @@
-    }
+  const platform = detectBrowserPlatform();
 
-    child.stdout.on('data', chunk => {
-      lines.push(chunk);
+  if (!platform) {
+    throw new Error('Unable to detect browser platform');
+  }
 
-      if (!returnValue) {
-        process.stdout.write(chunk);
+  // Get the version to install of both Chrome and Chromedriver
+  let buildId;
+  try {
+    buildId = await resolveBuildId(Browser.CHROME, platform, version);
+  } catch (e) {
+    throw new Error(e);
+  }
+
+  // Get any currently installed version
+  const currentVersion = getVersion();
+  const cacheDir = getBDMCacheDir();
+  if (currentVersion) {
+    if (currentVersion === buildId) {
+      console.log(
+        `Chrome and Chromedriver versions ${currentVersion} are already installed. Skipping installation.`
+      );
+      return;
+    }
+    console.log(
+      `Chrome and Chromedriver versions ${currentVersion} are currently installed. Overwriting.`
+    );
+    // Uninstall existing installs
+    ['chrome', 'chromedriver'].map(async browser => {
+      try {
+        await uninstall({
+          browser,
+          buildId: currentVersion,
+          cacheDir
+        });
+        fsPromises.rmdir(path, { recursive: true, force: true });
+      } catch (e) {
+        throw new Error(`Unable to remove ${browser}.`);
       }
     });
+    try {
+      // Remove the existing .env. setEnv creates it again later
+      // Should execution stop beforehand, .env will not be in a bad (empty) state
+      await fsPromises.rm(path.resolve(cacheDir, '.env'));
+    } catch (e) {
+      throw new Error(`Unable to remove .env from ${cacheDir}.`);
+    }
+  }
 
-    // for some reason the install script will output curl download
-    // percent through stderr
-    child.stderr.on('data', chunk => {
-      // mimic download percent bar by manually clearing lines
-      // @see https://stackoverflow.com/questions/17309749/node-js-console-log-is-it-possible-to-update-a-line-rather-than-create-a-new-l
-      if (chunk.indexOf('\r') !== -1) {
-        process.stdout.clearLine();
-      }
-      if (lastChunk) {
-        process.stdout.cursorTo(lastChunk.length - 1);
-      }
-      process.stdout.write(chunk);
+  // Create a cache directory if it does not exist on the user's home directory, or if it's been removed above
+  // This will be where environment variables will be stored for the tests
+  // since it is a consistent location across different platforms
+  try {
+    await fsPromises.mkdir(cacheDir, { recursive: true });
+  } catch {
+    console.log(`${cacheDir} already exists`);
+  }
 
-      lastChunk = chunk;
+  let installedChrome, installedChromedriver;
+  try {
+    installedChrome = await installBrowser(
+      cacheDir,
+      Browser.CHROME,
+      buildId,
+      options
+    );
+  } catch (e) {
+    e.message = /status code 404/.test(e.message)
+      ? `Unable to find version ${buildId} of Chrome`
+      : e.message;
+    throw new Error(e);
```

Comment:
You can just throw the original error instead of a new one

```suggestion
    throw e;
```